spec/policy/jwt_claim_check/jwt_claim_check_spec.lua

```diff
@@ -0,0 +1,286 @@
+        assert.same(ngx.status, 403)
+      end)
+
+    end)
+
+  end)
+
+  describe("validate methods entry", function()
+    it("Invalid method with valid operations", function()
+      local jwt_check = JWTClaimCheckPolicy.new({rules={
+          {
+            operations={{op="==", jwt_claim="{{foo}}", jwt_claim_type="liquid", value="fooValue"}},
+            combine_op="and",
+            methods  = {"POST"},
+            resource = ngx.var.uri
+          }
+        }})
+        jwt_check:access(context)
+        assert.not_same(ngx.status, 403)
+    end)
+
+    it("ANY method with valid operations", function()
+      local jwt_check = JWTClaimCheckPolicy.new({rules={
+          {
+            operations={{op="==", jwt_claim="{{foo}}", jwt_claim_type="liquid", value="fooValue"}},
+            combine_op="and",
+            methods  = {"ANY"},
+            resource = ngx.var.uri
+          }
+        }})
+        jwt_check:access(context)
+        assert.not_same(ngx.status, 403)
+    end)
+
+    it("ANY method with no matching JWT role operations", function()
+      local jwt_check = JWTClaimCheckPolicy.new({rules={
+          {
+            operations={{op="==", jwt_claim="{{ foo }}", jwt_claim_type="liquid", value="bar"}},
+            combine_op="and",
+            methods  = {"ANY"},
+            resource = ngx.var.uri
+          }
+        }})
+        jwt_check:access(context)
+        assert.same(ngx.status, 403)
+    end)
+
+    it("Multiple methods with matching JWT role operations", function()
+      local jwt_check = JWTClaimCheckPolicy.new({rules={
+          {
+            operations={{op="==", jwt_claim="{{ foo }}", jwt_claim_type="liquid", value="fooValue"}},
+            combine_op="and",
+            methods  = {"POST", "GET"},
+            resource = ngx.var.uri
+          }
+        }})
+        jwt_check:access(context)
+        assert.not_same(ngx.status, 403)
+
+        ngx.req.get_method:revert()
+        stub(ngx.req, 'get_method', function() return 'POST' end)
+        jwt_check:access(context)
+        assert.not_same(ngx.status, 403)
+
+        -- PUT method is not part of the rule, so it's allowed.
+        ngx.req.get_method:revert()
+        stub(ngx.req, 'get_method', function() return 'PUT' end)
+        jwt_check:access(context)
+        assert.not_same(ngx.status, 403)
+    end)
+
+  end)
+
+  it("validates custom error message", function()
+      local message = "custom one"
+      spy.on(ngx, "say")
+      local jwt_check = JWTClaimCheckPolicy.new({
+        rules={
+          {
+            operations={{op="!=", jwt_claim="{{ foo }}", jwt_claim_type="liquid", value="fooValue"}},
+            combine_op="and",
+            methods  = {"POST", "GET"},
+            resource = ngx.var.uri
+          }
+        },
+        error_message=message})
+      jwt_check:access(context)
+      assert.same(ngx.status, 403)
+      assert.spy(ngx.say).was.called()
+      assert.spy(ngx.say).was.called_with(message)
```

Comment:
This one makes the previous line redundant, right?

Reply:
I really liked this one, because can be called without arguments, so if that happens we know at least that was called.

Reply:
:+1: